examples/da/plot_otda_d2.py

```diff
@@ -0,0 +1,163 @@
+# -*- coding: utf-8 -*-
+"""
+==============================
+OT for empirical distributions
+==============================
+
+This example introduces a domain adaptation in a 2D setting. It explicits
+the problem of domain adaptation and introduces some optimal transport
+approaches to solve it.
+
+Quantities such as optimal couplings, greater coupling coefficients and
+transported samples are represented in order to give a visual understanding
+of what the transport methods are doing.
+"""
+
+# Authors: Remi Flamary <remi.flamary@unice.fr>
+#          Stanislas Chambon <stan.chambon@gmail.com>
+#
+# License: MIT License
+
+import matplotlib.pylab as pl
+import ot
+
+# number of source and target points to generate
+ns = 150
+nt = 150
+
+Xs, ys = ot.datasets.get_data_classif('3gauss', ns)
+Xt, yt = ot.datasets.get_data_classif('3gauss2', nt)
+
+# Cost matrix
+M = ot.dist(Xs, Xt)
```

Comment:
if it's doing euclidian distance than naming / API is not explicit enough IMO

Reply:
dist is basically a wrapper around scipy cdist where you can choose the metric.

Also it shouldn't be computed here since it's not used anywhere it seems.

Reply:
Actually I did compute it to plot it along with source and target points